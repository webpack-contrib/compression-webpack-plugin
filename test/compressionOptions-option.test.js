import Plugin from '../src/index';

import {
  compile,
  getAssetsNameAndSize,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers/index';

describe('"compressionOptions" option', () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler('./entry.js');
  });

  it('matches snapshot without values', async () => {
    new Plugin({
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for custom options ({Object})', async () => {
    new Plugin({
      compressionOptions: {
        level: 1,
      },
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });
});
