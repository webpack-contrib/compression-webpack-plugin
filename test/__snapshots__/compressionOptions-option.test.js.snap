// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`compressionOptions\` option matches snapshot for custom options ({Object}): assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js.gz",
    117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73332,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    404,
  ],
  Array [
    "js.b329671af0217355535e.js",
    10267,
  ],
  Array [
    "js.b329671af0217355535e.js.gz",
    3180,
  ],
]
`;

exports[`when applied with \`compressionOptions\` option matches snapshot for custom options ({Object}): errors 1`] = `Array []`;

exports[`when applied with \`compressionOptions\` option matches snapshot for custom options ({Object}): warnings 1`] = `Array []`;

exports[`when applied with \`compressionOptions\` option matches snapshot without values: assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js.gz",
    117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.b329671af0217355535e.js",
    10267,
  ],
  Array [
    "js.b329671af0217355535e.js.gz",
    2798,
  ],
]
`;

exports[`when applied with \`compressionOptions\` option matches snapshot without values: errors 1`] = `Array []`;

exports[`when applied with \`compressionOptions\` option matches snapshot without values: warnings 1`] = `Array []`;
